src/box/lua/space.cc

```diff
@@ -279,6 +279,120 @@ lbox_push_space_foreign_key(struct lua_State *L, struct space *space, int i)
 	lua_setfield(L, i, "foreign_key");
 }
 
+/**
+ * index.parts:extract_key(tuple)
+ * Stack: [1] unused; [2] tuple.
+ * key_def is passed in the upvalue.
+ */
+static int
+lbox_key_def_extract_key(struct lua_State *L)
+{
+	struct key_def *key_def = luaT_check_key_def(L, lua_upvalueindex(1));
+	if (lua_gettop(L) != 2 || key_def == NULL)
+		return luaL_error(L, "Usage: index.parts:extract_key(tuple)");
+	if (key_def->is_multikey)
+		return luaL_error(L, "multikey path is unsupported");
+	return luaT_key_def_extract_key(L, key_def, 2);
+}
+
+/**
+ * index.parts:compare(tuple_a, tuple_b)
+ * Stack: [1] unused; [2] tuple_a; [3] tuple_b.
+ * key_def is passed in the upvalue.
+ */
+static int
+lbox_key_def_compare(struct lua_State *L)
+{
+	struct key_def *key_def = luaT_check_key_def(L, lua_upvalueindex(1));
+	if (lua_gettop(L) != 3 || key_def == NULL) {
+		return luaL_error(L, "Usage: index.parts:compare("
+				     "tuple_a, tuple_b)");
+	}
+	if (key_def->is_multikey)
+		return luaL_error(L, "multikey path is unsupported");
+	return luaT_key_def_compare(L, key_def, 2, 3);
+}
+
+/**
+ * index.parts:compare_with_key(tuple, key)
+ * Stack: [1] unused; [2] tuple; [3] key.
+ * key_def is passed in the upvalue.
+ */
+static int
+lbox_key_def_compare_with_key(struct lua_State *L)
+{
+	struct key_def *key_def = luaT_check_key_def(L, lua_upvalueindex(1));
+	if (lua_gettop(L) != 3 || key_def == NULL) {
+		return luaL_error(L, "Usage: index.parts:compare_with_key("
+				     "tuple, key)");
+	}
+	if (key_def->is_multikey)
+		return luaL_error(L, "multikey path is unsupported");
+	return luaT_key_def_compare_with_key(L, key_def, 2, 3);
+}
+
+/**
+ * index.parts:merge(second_key_def)
+ * Stack: [1] unused; [2] second_key_def.
+ * key_def is passed in the upvalue.
+ */
+static int
+lbox_key_def_merge(struct lua_State *L)
+{
+	struct key_def *key_def_a = luaT_check_key_def(L, lua_upvalueindex(1));
+	struct key_def *key_def_b = luaT_check_key_def(L, 2);
```

Comment:
Shouldn't it accept another index parts here?

Reply:
OK, done.